### PR TITLE
 Issue #5124: remove usages of branchContains for EqualsAvoidNullCheck

### PR DESCRIPTION
**Pull request by pbludov**
Issue #5124

This is just a simple rewrite.
Regression: https://pbludov.github.io/issue-5124/ (empty)

### CODE REVIEW

**Comment by rnveach** [comment]
commit validation failed, looks like there is a space before the first word of the commit message.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java [hunk 1]
```diff
@@ -351,9 +351,39 @@ private static boolean containsAllSafeTokens(final DetailAST expr) {
...
+        return argIsNotNull || !branchContainsIdentOrNull(arg);
+    }
+
+    /**
+     * Checks whether a {@code null} or {@code ident} exist under token.
+     *
+     * @param token tree token.
+     * @return true if a {@code null} or {@code ident} exists under token.
+     */
+    private static boolean branchContainsIdentOrNull(DetailAST token) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java]: This is basically just a re-hash of the original `branchContains` method. There is no other way to look at what was intended and rewrite in some other fashion?
Like what are we looking for by `IDENT`? A variable, field, methods, etc or any of the above?

**Comment by romani** [comment]
@pbludov , @rnveach , @strkkk 

TC validation is slow again. 40 minutes is not enough. I updated TC to timeout in 50 min:
image

**Comment by romani** [comment]
Travis failure was:
```
[INFO] Running com.puppycrawl.tools.checkstyle.internal.CommitValidationTest

[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.231 s <<< FAILURE! - in com.puppycrawl.tools.checkstyle.internal.CommitValidationTest

[ERROR] testCommitMessageHasProperStructure(com.puppycrawl.tools.checkstyle.internal.CommitValidationTest)  Time elapsed: 0.007 s  <<< FAILURE!

java.lang.AssertionError: 

Commit 497bd5e50d04e60a82a2ef34b73f92b78f37c94e message: " Issue #5124: remove usages of branchContains for EqualsAvoidNullCheck\n" is invalid

Proper commit message should adhere to the following rules:

    1) Must match one of the following patterns:

        ^Issue #\d+: .*$

        ^Pull #\d+: .*$

        ^(minor|config|infra|doc|spelling): .*$

    2) It contains only one line of text

    3) Must not end with a period, space, or tab

    4) Commit message should be less than or equal to 200 characters

The rule broken was: 1
```

strange, message looks fine .... restarted.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java (hunk 1 above)
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java]: You are right. This method is redundant here. To make the long story short, is handles 
```
        s.equalsIgnoreCase(((s = "cold pizza")));
```
but this should be done in the method `skipVariableAssign`.

**Pull request closed without merging**